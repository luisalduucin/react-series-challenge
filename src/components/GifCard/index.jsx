import React from 'react';
import PropTypes from 'prop-types';
import {Btn, BtnsWrapper, GifCardWrapper} from "./GifCard.style";
import eyeIcon from './assets/eye.png';
import heartBorder from './assets/heart-border.svg';
import heartFilled from './assets/heart-filled.svg';

const GifCard = ({gif, isFavorite, imageSize, toggleFavorite}) => {
  const image = imageSize === 'small' ? gif.images.small : gif.images.original;
  const heartIcon = isFavorite ? heartFilled : heartBorder;
  return (
    <GifCardWrapper>
      <img alt={'gif'} src={image} />
      <BtnsWrapper>
        <Btn alt={'open'} src={eyeIcon} />
        <Btn alt={'favorite'} src={heartIcon} onClick={toggleFavorite.bind(this, gif)} />
      </BtnsWrapper>
    </GifCardWrapper>
  )
};

GifCard.propTypes = {
  gif: PropTypes.shape({
    id: PropTypes.string,
    images: PropTypes.shape({
      original: PropTypes.string,
      small: PropTypes.string
    })
  }).isRequired,
  isFavorite: PropTypes.bool.isRequired,
  imageSize: PropTypes.oneOf(['small', 'original']).isRequired,
  toggleFavorite: PropTypes.func.isRequired
};

export default GifCard
