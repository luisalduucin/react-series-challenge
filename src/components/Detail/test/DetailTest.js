import React from 'react';
import {Detail} from "../index";
import {shallow} from "enzyme";
import {GifCard} from "../../GifCard";

const paramGifId = {params: {id: 'p3n7NWvfz3u0c1meay'}};
const emptyFavorites = {
  original: [],
  filtered: []
};
const emptyTrending = [];
const mockGif = {
  id: 'p3n7NWvfz3u0c1meay',
  images: {
    original: 'https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif',
    small: 'https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif'
  }
};
const mockFavorites = {
  original: [mockGif],
  filtered: [mockGif]
};

describe('Detail component', () => {
  const func = jest.fn();
  it('should render component', () => {
    const component = shallow(
      <Detail trending={emptyTrending} favorites={emptyFavorites} match={paramGifId}/>
    );
    expect(component.contains(
      <GifCard
        gif={mockGif}
        isFavorite={true}
        isDetailBtnVisible={false}
        imageSize={'original'}
        toggleFavorite={func}
      />
    )).toEqual(false);
  });

  it('should render component', () => {
    const component = shallow(
      <Detail trending={emptyTrending} favorites={mockFavorites} match={paramGifId}/>
    );
    expect(component).toHaveLength(1);
  });
});