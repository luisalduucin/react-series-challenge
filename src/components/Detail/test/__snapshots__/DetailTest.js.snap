// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail component should render component 1`] = `
<div
  className="sc-bdVaJa cbIwnt"
>
  <div
    style={Object {}}
  >
    <p
      className="sc-bwzfXH otGqO"
    >
      Detail
    </p>
    <img
      className="sc-bxivhb gUOyqx"
      onClick={[Function]}
      src={
        Object {
          "process": [Function],
        }
      }
    />
    <img
      className="sc-bxivhb gUOyqx"
      onClick={[Function]}
      src={
        Object {
          "process": [Function],
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "display": "inline-flex",
        "marginTop": "15px",
      }
    }
  >
    <div
      className="sc-gzVnrw hLMxiO"
    >
      <p
        className="sc-dnqmqq dysggb"
      >
        404
      </p>
      <p
        className="sc-iwsKbI fKNdwq"
      >
        The requested url was not found
      </p>
    </div>
  </div>
</div>
`;

exports[`Detail component should render component 2`] = `
<div
  className="sc-bdVaJa cbIwnt"
>
  <div
    style={Object {}}
  >
    <p
      className="sc-bwzfXH otGqO"
    >
      Detail
    </p>
    <img
      className="sc-bxivhb gUOyqx"
      onClick={[Function]}
      src={
        Object {
          "process": [Function],
        }
      }
    />
    <img
      className="sc-bxivhb gUOyqx"
      onClick={[Function]}
      src={
        Object {
          "process": [Function],
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "display": "inline-flex",
        "marginTop": "15px",
      }
    }
  >
    <div
      className="sc-gzVnrw hLMxiO"
    >
      <p
        className="sc-dnqmqq dysggb"
      >
        404
      </p>
      <p
        className="sc-iwsKbI fKNdwq"
      >
        The requested url was not found
      </p>
    </div>
  </div>
</div>
`;

exports[`Detail component should render component 3`] = `
<div
  className="sc-bdVaJa cbIwnt"
>
  <div
    style={Object {}}
  >
    <p
      className="sc-bwzfXH otGqO"
    >
      Detail
    </p>
    <img
      className="sc-bxivhb gUOyqx"
      onClick={[Function]}
      src={
        Object {
          "process": [Function],
        }
      }
    />
    <img
      className="sc-bxivhb gUOyqx"
      onClick={[Function]}
      src={
        Object {
          "process": [Function],
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "display": "inline-flex",
        "marginTop": "15px",
      }
    }
  >
    <div
      className="sc-gzVnrw hLMxiO"
    >
      <p
        className="sc-dnqmqq dysggb"
      >
        404
      </p>
      <p
        className="sc-iwsKbI fKNdwq"
      >
        The requested url was not found
      </p>
    </div>
  </div>
</div>
`;
